src/h2olog/misc/gen_raw_tracer.py

```diff
@@ -423,13 +434,29 @@ def build_tracer(context, metadata):
 #endif
 """
 
-  c += r"""
+  if fully_specified_probe_name == "h2o:socket_lookup_flags":
+    c += r"""
+#ifdef H2OLOG_SAMPLING_RATE
+  uint64_t flags = event.socket_lookup_flags.original_flags;
+  int skip_tracing = bpf_get_prandom_u32() > (H2OLOG_SAMPLING_RATE * U32_MAX);
```

Comment:
IMHO: eBPF does not support floating point number yet. I suppose that using floating point number in BPF code is not good(in general, floating point number is not used in kernel space). Well, this code does not have problem because a compiler can calculate its value in compile time and generate it as integer value. I suppose percentage(0-100) is better than rate(0.0 - 1.0) as below.

```c
int skip_tracing = bpf_get_prandom_u32() > (U32_MAX / 100 * PERCENTAGE)
```

Reply:
Oh, good catch! 

bpf(2) manpage says:

> eBPF programs can be written in a restricted C that is compiled (using the clang compiler) into eBPF bytecode. Various features are omitted from this restricted C, such as loops,  global variables,  variadic functions,  floating-point numbers, and passing structures as function arguments.  Some examples can be found in the samples/bpf/*_kern.c files in the kernel source tree.

I'm not sure the percentage precision is good enough, so I'll calculate `H2OLOG_SAMPLING_RATE * U32_MAX)` in the userspace.